src/components/layout/Navbar.tsx

```diff
@@ -0,0 +1,97 @@
+import { Link } from 'react-router-dom';
+import { useRef, useState, useEffect, useContext } from 'react';
+import { AuthContext } from '@/context/AuthContext';
+import NotificationModal from '../common/notification-modal/NotificationModal';
+import logo from '@/assets/images/logo.svg';
+import search from '@/assets/icons/search.svg';
+import alarmActive from '@/assets/icons/alarm-active.svg';
+import alarmInactive from '@/assets/icons/alarm-inactive.svg';
+
+export default function Navbar() {
+  const { isLoggedIn, role, alarms, logout } = useContext(AuthContext);
+  const [isShowModal, setIsShowModal] = useState(false);
+  const modalRef = useRef<HTMLDivElement>(null);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+
+  // 바깥 클릭 시 모달 닫기
+  useEffect(() => {
+    function handleClickOutside(event: MouseEvent): void {
+      const target = event.target as Node
+      if (
+        isShowModal &&
+        modalRef.current &&
+        !(modalRef.current as HTMLElement).contains(target) &&
+        buttonRef.current &&
+        !(buttonRef.current as HTMLElement).contains(target)
```

Comment:
```suggestion
        modalRef.current &&
        !modalRef.current.contains(target) &&
        buttonRef.current &&
        !buttonRef.current.contains(target)
```
❗ 이미 앞에서 `null`에 대한 검정을 해줘서 타입 단언을 빼도 오류 없이 작동하는 거 같습니다!